vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/screenshot/ScreenCapturerTestRunListener.kt

```diff
@@ -88,9 +89,13 @@ class ScreenCapturerTestRunListener(
         /**
          * We might not observe the testEnded event, but the testRunFailed will always be reported
          */
-        supervisorJob?.let {
-            logger.debug { "Closing dangling screencapturer job" }
-            it.cancel()
+        supervisorJob?.cancelAndJoin()
+        lastTestIdentifier?.let { id ->
+            val existingRecording = fileManager.createFile(FileType.SCREENSHOT, pool, device.toDeviceInfo(), id.toTest(), testBatchId)
+            if (existingRecording.length() > 0) {
```

Comment:
we don't need to check if device is healthy here since the local gif is either created or not